### PR TITLE
Overall improvements and bugfixes

### PR DESCRIPTION
**Pull request by Wouter0100**
A bunch of bugfixes, improvements and docblock update's :)

### CODE REVIEW

File: FileManager/src/Model/Table/AttachmentsTable.php [hunk 1]
```diff
@@ -57,12 +57,18 @@ public function __construct(array $config = [])
     protected function _saveUploadedFile($data)
     {
         $file = $data->file;
+        $dir = WWW_ROOT . $this->uploadsDir;
+
+        // Check if dir exists
+        if (!file_exists($dir)) {
```
Comment (Wouter0100) [inline FileManager/src/Model/Table/AttachmentsTable.php]: Or should we only do this when `debug` is enabled?

Comment (Marlinc) [inline FileManager/src/Model/Table/AttachmentsTable.php]: I do think it makes sense to be consistent with the framework.

Comment (Wouter0100) [inline FileManager/src/Model/Table/AttachmentsTable.php]: Added.

**Comment by Wouter0100** [comment]
This PR should be ready for merge. Many improvements across whole Croogo are made so you'll get less errors and so on :).

**Comment by Marlinc** [comment]
Looking good!

**Comment by Marlinc** [comment]
Thanks!

**Pull request merged**